packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -236,6 +236,15 @@ export const Table = memo((props: Props) => {
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  useEffect(() => {
+    // Reset page index when data changes
+    // This is needed because react-table does not do this automatically
+    // autoResetPage is set to false because setting it to true causes the issue described in
+    // https://github.com/grafana/grafana/pull/67477
+    gotoPage(0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data]);
```

Comment:
Are we sure that we want to force user to first page on _every_ data change? 🤔 in case of streaming that would not be ideal... I think we may need to do some in depth testing here 😬

Reply:
For me, it makes sense to switch to the first page when new data arrives, but maybe there a some cases.
I don't know how streaming works, can we check it together?

Reply:
I'm also not familiar with how streaming works, but I would guess every time new data comes in, React would trigger a re-render, as well as this pagination reset. Maybe only conditionally reseting the page if NOT streaming?

Reply:
Hm, in this case I'll add some condition, no problem.

Reply:
I'd say we want to go to the first page when the current page is no longer existing (i.e. there's not enough data) and otherwise we stay on the current page if there is enough data.

Granted we may want to figure out a better strategy for pagination and streaming. It seems like it's almost guaranteed to get wacky results in that case.